aws_lambda_powertools/shared/functions.py

```diff
@@ -2,14 +2,14 @@
 from typing import Any, Optional, Union
 
 
-def resolve_truthy_env_var_choice(env: Any, choice: bool = None) -> bool:
+def resolve_truthy_env_var_choice(env: str, choice: Optional[bool] = None) -> bool:
     """Pick explicit choice over truthy env value, if available, otherwise return truthy env value
 
     NOTE: Environment variable should be resolved by the caller.
 
     Parameters
     ----------
-    env : Any
+    env : str
```

Comment:
it surprises me it didn't suggest Optional[str] as `env` could be None if we don't call `getenv` with a default. I suspect it scanned the code to see how it was called dunno

Reply:
`strtobool(env)` would fail if `env` is not set (val.lower())

```python
def strtobool (val):
    """Convert a string representation of truth to true (1) or false (0).
    True values are 'y', 'yes', 't', 'true', 'on', and '1'; false values
    are 'n', 'no', 'f', 'false', 'off', and '0'.  Raises ValueError if
    'val' is anything else.
    """
    val = val.lower()
    if val in ('y', 'yes', 't', 'true', 'on', '1'):
        return 1
    elif val in ('n', 'no', 'f', 'false', 'off', '0'):
        return 0
    else:
        raise ValueError("invalid truth value %r" % (val,))
```

Reply:
@heitorlessa see comment

Reply:
Ah yes! That's why I had set as Any at first I think as I was playing with having error handling or good defaults